public/client_example/trial_analysis_YorkAT.rb

```diff
@@ -0,0 +1,193 @@
+  response = call_bip request
+  break if response['plant_scoring_units'].size == 0
+  response['plant_scoring_units'].each do |plant_scoring_unit|
+    outputs[plant_scoring_unit['scoring_unit_name']] = {}
+    outputs[plant_scoring_unit['scoring_unit_name']]['plant_accession_id'] = plant_scoring_unit['plant_accession_id']
+    plant_accession_ids << plant_scoring_unit['plant_accession_id']
+    outputs[plant_scoring_unit['scoring_unit_name']]['trait_scores'] =
+      trait_scores.select{ |ts| ts['plant_scoring_unit_id'] == plant_scoring_unit['id']}
+  end
+  page += 1
+  plant_scoring_units += response['plant_scoring_units']
+end
+plant_accession_ids.uniq!
+
+STDERR.puts "Number of unique PA ids : #{plant_accession_ids.length}"
+
+
+STDERR.puts '5. Finding Plant Accessions for this Plant Trial.'
+
+plant_accessions = []
+plant_accession_ids.each_slice(200) do |plant_accession_ids_slice|
+  ids_pa_param = plant_accession_ids_slice.map{ |pa_id| "plant_accession[query][id][]=#{pa_id}" }.join("&")
+  request = Net::HTTP::Get.new("/api/v1/plant_accessions?#{ids_pa_param}&per_page=200", @headers)
+  response = call_bip request
+  plant_accessions += response['plant_accessions']
+  STDERR.print '.'
+end
+
+STDERR.puts "  - The Plant Accessions used in this Plant Trial: #{plant_accessions.map{ |pa| pa['plant_accession'] }.join(", ")}"
+STDERR.puts "Number of Plant Accessions loaded: #{plant_accessions.size}"
+
+
+STDERR.puts '6. Finding Plant Lines for this Plant Trial.'
+
+plant_line_ids = plant_accessions.map{ |pa| pa['plant_line_id'] }.uniq
+plant_lines = []
+plant_line_ids.each_slice(200) do |plant_line_ids_slice|
+  ids_pl_param = plant_line_ids_slice.map{ |pl_id| "plant_line[query][id][]=#{pl_id}" }.join("&")
+  request = Net::HTTP::Get.new("/api/v1/plant_lines?#{ids_pl_param}&per_page=200", @headers)
+  response = call_bip request
+  plant_lines += response['plant_lines']
+end
+
+STDERR.puts "  - The Plant Lines used in this Plant Trial: #{plant_lines.map{ |pl| pl['plant_line_name'] }.join(", ")}"
+STDERR.puts "Number of Plant Lines loaded: #{plant_lines.size}"
+
+
+STDERR.puts "7. Linking Plant Accesions with Plant Scoring Units"
+
+page = 1
+loop do
+  request = Net::HTTP::Get.new("/api/v1/plant_scoring_units?plant_scoring_unit[query][plant_trials.id]=#{plant_trial_id}&page=#{page}&per_page=200", @headers)
+  response = call_bip request
+  break if response['plant_scoring_units'].size == 0
+  response['plant_scoring_units'].each do |plant_scoring_unit|
+    outputs[plant_scoring_unit['scoring_unit_name']]['plant_accession'] = plant_accessions.detect { |pa| pa['id'] == plant_scoring_unit['plant_accession_id']}
+  end
+  page += 1
+end
+
+
+STDERR.puts '8. Generating output TSV to STDOUT'
+
+output_string = CSV.generate(col_sep: "\t") do |csv|
+  csv << ["<Trait>"] + trait_descriptors.map{ |td| td['trait']['name'].gsub(/\s+/, '.') }
```

Comment:
This is based on an example file I was given by Wiktor. I do not have any more information, so I just tried to make the output as similar as possible. I will add some comments regarding this formatting.